pkg/client/block/block.go

```diff
@@ -26,19 +28,23 @@ func (blockEvent *cometBlockEvent) Hash() []byte {
 	return blockEvent.Block.LastBlockID.Hash.Bytes()
 }
 
-// newCometBlockEvent attempts to deserialize the given bytes into a comet block.
+// newCometBlockEventFactoryFn is a factory function that returns a functon
+// that attempts to deserialize the given bytes into a comet block.
 // if the resulting block has a height of zero, assume the event was not a block
 // event and return an ErrUnmarshalBlockEvent error.
-func newCometBlockEvent(blockMsgBz []byte) (client.Block, error) {
-	blockMsg := new(cometBlockEvent)
-	if err := json.Unmarshal(blockMsgBz, blockMsg); err != nil {
-		return nil, err
+func newCometBlockEventFactoryFn(ctx context.Context) events.NewEventsFn[client.Block] {
```

Comment:
Why does this function need to close over the `ctx`? It doesn't look like the returned function uses it at all. :thinking: